package-res/resources/web/pentaho/type/property.js

```diff
@@ -0,0 +1,1331 @@
+           * However, a _property type_ marked as _not applicable_ can no longer go back to being _applicable_.
+           *
+           * Because this attribute is also _dynamic_,
+           * the actual `isApplicable` values are only known
+           * when evaluated for specific complex instances.
+           * As such, ensuring monotonic changes is deferred until evaluation.
+           * No errors are thrown; non-monotonic changes simply don't take any effect.
+           *
+           * ### This attribute is *Inherited*
+           *
+           * When there is no _local value_, the _effective value_ of the attribute is the _inherited effective value_.
+           *
+           * The first set local value must respect the _monotonicity_ property with the inherited value.
+           *
+           * ### Other characteristics
+           *
+           * The value got by the attribute is the **last set local, value**, if any -
+           * a function, a constant value or `undefined`, when unset.
+           *
+           * When set to a {@link Nully} value, the set operation is ignored.
+           *
+           * When set and the property already has [descendant]{@link pentaho.type.Type#hasDescendants} properties,
+           * an error is thrown.
+           *
+           * The default (root) `isApplicable` attribute value is `true`.
+           *
+           * @name isApplicable
+           * @memberOf pentaho.type.Property.Type#
+           * @type undefined | boolean | pentaho.type.PropertyDynamicAttribute.<boolean>
+           *
+           * @see pentaho.type.Property.Type#isRequired
+           * @see pentaho.type.Complex#isApplicable
+           * @see pentaho.type.spec.IPropertyTypeProto#isApplicable
+           */
+          isApplicable: {
+            value: true,
+            cast: Boolean,
+            combine: function(baseEval, localEval) {
+              return function() {
+                // localEval is skipped if base is false.
+                return baseEval.call(this) && localEval.call(this);
+              };
+            }
+          },
+
+          /**
+           * Evaluates the value of the `isReadOnly` attribute of a property of this type
+           * on a given owner complex value.
+           *
+           * @name isReadOnlyEval
+           * @memberOf pentaho.type.Property.Type#
+           * @param {pentaho.type.Complex} owner The complex value that owns a property of this type.
+           * @return {boolean} The evaluated value of the `isReadOnly` attribute.
+           *
+           * @ignore
+           */
+
+          /**
+           * Gets or sets a value, or function, that indicates if properties of this type
+           * _cannot_ be changed by a user, in a user interface.
+           *
+           * A property should be set read-only whenever its value is implied/imposed somehow,
+           * and thus cannot not be changed, directly, by the user, in a user interface.
```

Comment:
Suggest:

and thus cannot not be changed directly by the user through a user interface.